File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
    
    int maxVowels(string s, int k) {
        int n = s.size(), vowels = 0;
        for (int i = 0; i < k; i++) {
            if  (isVowel(s[i])) {
                vowels++;
            }
        }
        int res = vowels;
        for (int i = k; i < n; i++) {
            if (isVowel(s[i - k])) {
                vowels--;
            }
            if (isVowel(s[i])) {
                vowels++;
            }
            res = max(res, vowels);
        }
        return res;
    }
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public static String vowels = "aeiou";
    
    public int maxVowels(String s, int k) {
        int currVowelCount = 0;
        for (int i = 0; i < k; i++) {
            if (isVowel(s.charAt(i))) {
                currVowelCount++;
            }
        }
        int maxVowelCount = currVowelCount;
        for (int i = k; i < s.length(); i++) {
            if (isVowel(s.charAt(i))) {
                currVowelCount++;
            }
            if (isVowel(s.charAt(i - k))) {
                currVowelCount--;
            }
            maxVowelCount = Math.max(maxVowelCount, currVowelCount);
            if (maxVowelCount == k) {
                return k;
            }
        }
        return maxVowelCount;
    }
    
    public boolean isVowel(char c) {
        return vowels.indexOf(c) != -1;
    }
}